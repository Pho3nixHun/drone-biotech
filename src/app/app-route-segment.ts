export enum AppRouteSegment {
  PRODUCT = 'products',
  PARTNERS = 'partners',
}
