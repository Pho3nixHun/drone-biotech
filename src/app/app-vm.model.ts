import { LogoVM } from '@components/header/components/logo/logo-vm.model';
import { NavItemVM } from '@components/header/components/nav/components/nav-item/nav-item-vm';
import { WithRouterLink } from '@interfaces/with-router-link.interface';
import { WithTextNode } from '@interfaces/with-text-node.interface';

type NavItemXVM = NavItemVM & WithTextNode;
type LogoXVM = WithRouterLink & LogoVM;
export interface AppComponentVM {
    headerCanBeShown: boolean;
    logoXVM: LogoXVM;
    navItemXVMs: NavItemXVM[];
}
