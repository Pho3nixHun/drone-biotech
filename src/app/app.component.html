<div class="min-h-screen flex flex-col">
  <app-header>
    <app-logo [vm]="logoVM()"></app-logo>
    <app-nav>
      <a href="" class="sm:hover:text-gray-400 duration-150 cursor-pointer">Home</a>
      <a class="sm:hover:text-gray-400 duration-150 cursor-pointer">Products</a>
      <a class="sm:hover:text-gray-400 duration-150 cursor-pointer">Partners</a>
      <a class="sm:hover:text-gray-400 duration-150 cursor-pointer">Contacts</a>
    </app-nav>
  </app-header>
  <router-outlet class="flex-grow"></router-outlet>
</div>
