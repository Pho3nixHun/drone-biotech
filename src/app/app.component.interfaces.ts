import { LogoComponentInterfaces } from '@components/header/components/logo/logo.component.interfaces';

export interface AppComponentInterfaces {
  logoVM: LogoComponentInterfaces;
}
