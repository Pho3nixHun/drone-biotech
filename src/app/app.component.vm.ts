import { LogoComponentInterfaces } from '@components/header/components/logo/logo.component.interfaces';

export interface AppComponentVM {
  logoVM: LogoComponentInterfaces;
}
