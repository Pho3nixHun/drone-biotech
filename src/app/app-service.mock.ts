import { computed, inject, Injectable, Signal, signal } from '@angular/core';
import { AppComponentVM } from './app-vm.model';
import { AppService } from './app.service';
import { toSignal } from '@angular/core/rxjs-interop';
import { selectHeaderCanBeShown } from './stores/router/router.selectors';
import { Store } from '@ngrx/store';

const getVMSignal = signal<AppComponentVM | undefined>(undefined);

@Injectable({
    providedIn: 'root',
})
class AppComponentMockService {
    private readonly headerCanBeShown = toSignal(
        inject(Store).select(selectHeaderCanBeShown),
        { initialValue: false }
    );

    private readonly vm = computed<AppComponentVM | undefined>(() => {
        const vmSignal = getVMSignal();
        if (!vmSignal) {
            return vmSignal;
        }

        const headerCanBeShown = this.headerCanBeShown();
        return headerCanBeShown
            ? {
                  ...vmSignal,
                  headerCanBeShown: headerCanBeShown,
              }
            : vmSignal;
    });

    public getVM(): Signal<AppComponentVM | undefined> {
        return this.vm;
    }
}

export const provideAppComponentMockService = () => ({
    provide: AppService,
    useClass: AppComponentMockService,
});

export const updateGetVMSignal = getVMSignal.set.bind(getVMSignal);
