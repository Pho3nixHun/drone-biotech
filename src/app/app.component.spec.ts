import { ComponentFixture, TestBed } from '@angular/core/testing';
import { provideRouter } from '@angular/router';
import { AppComponent } from './app.component';
import { getTranslocoModule } from 'transloco-testing.module';
import { routes } from './app.routes';
import {
    enAppMock,
    appMockVM,
    appMockVMWithFiveNavItem,
    appMockVMWithOneNavItem,
    appMockVMWithoutNavItem,
    appMockVMWithOneAnchor,
    appMockVMWithFiveAnchor,
    appMockVMWithNavItemAndAnchor,
} from './app.mock';
import {
    provideAppComponentMockService,
    updateGetVMSignal,
} from './app-service.mock';
import { provideAuthenticatedUserGuard } from '@guards/authenticated-user/authenticated-user.guard.mock';
import { provideGuestOnlyGuard } from '@guards/guest-only/guest-only.guard.mock';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { selectHeaderCanBeShown } from '@stores/router/router.selectors';

describe('AppComponent', () => {
    let fixture: ComponentFixture<AppComponent>;
    let compiled: HTMLElement;
    let mockStore: MockStore;
    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                AppComponent,
                getTranslocoModule({
                    langs: { en: enAppMock },
                    translocoConfig: {
                        availableLangs: ['en'],
                        defaultLang: 'en',
                    },
                }),
            ],
            providers: [
                provideAppComponentMockService(),
                provideRouter(routes),
                provideGuestOnlyGuard(),
                provideAuthenticatedUserGuard(),
                provideMockStore({
                    selectors: [
                        { selector: selectHeaderCanBeShown, value: false },
                    ],
                }),
            ],
        }).compileComponents();

        fixture = TestBed.createComponent(AppComponent);
        mockStore = TestBed.inject(MockStore);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it(`should not render the template when the VM is not provided`, () => {
        //Arrange
        updateGetVMSignal(undefined);

        //Act
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });

    //Snapshot testing
    it(`should render the template when the VM is provided`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, true);
        updateGetVMSignal(appMockVM);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });

    //Snapshot test
    it(`should not render the <app-header> when the headerCanBeShown$ value is false`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, false);
        updateGetVMSignal(appMockVM);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });

    //Snapshot test
    it(`should render the <app-header> when the headerCanBeShown$ value is true`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, true);
        updateGetVMSignal(appMockVM);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });

    //Snapshot test
    it(`should not render <app-nav-item> when there is no item provided`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, true);
        updateGetVMSignal(appMockVMWithoutNavItem);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });

    //Snapshot test
    it(`should render 1 <app-nav-item> when 1 item is provided`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, true);
        updateGetVMSignal(appMockVMWithOneNavItem);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });
    //Snapshot test
    it(`should render 5 <app-nav-item> in order when 5 items are provided`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, true);
        updateGetVMSignal(appMockVMWithFiveNavItem);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });

    //Snapshot test
    it(`should render 1 <a> with routerLink when 1 item is provided`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, true);
        updateGetVMSignal(appMockVMWithOneAnchor);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });
    //Snapshot test
    it(`should render 5 <a> with routerLink in order when 5 items are provided`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, true);
        updateGetVMSignal(appMockVMWithFiveAnchor);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });
    //Snapshot test
    it(`should render <app-nav-item> and <a> with routerLink in order when both of them are provided`, () => {
        //Arrange
        mockStore.overrideSelector(selectHeaderCanBeShown, true);
        updateGetVMSignal(appMockVMWithNavItemAndAnchor);

        //Act
        mockStore.refreshState();
        fixture.detectChanges();

        //Assert
        expect(compiled).toMatchSnapshot();
    });
});
