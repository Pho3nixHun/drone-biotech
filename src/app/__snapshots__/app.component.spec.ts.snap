// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should navigate to "" and render the LandingPageComponent 1`] = `
<div
  id="root4"
  ng-version="18.2.8"
>
  <div
    class="min-h-screen flex flex-col"
  >
    <app-header>
      <section
        class="p-4 flex justify-center"
      >
        <div
          class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
        >
          <app-logo
            ng-reflect-vm="[object Object]"
          >
            <a
              href=""
            >
              <img
                alt="logo"
                class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
                src="assets/phoenix.jpg"
              />
            </a>
          </app-logo>
          <app-nav>
            <nav
              class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
            >
              <!--container-->
            </nav>
          </app-nav>
        </div>
      </section>
    </app-header>
    <!--container-->
    <div
      class="flex-grow"
    >
      <router-outlet />
      <app-landing-page>
        <app-hero
          ng-reflect-vm="[object Object]"
        >
          <section
            class="mt-10 h-[75vh] flex flex-col justify-center bg-no-repeat bg-right-top bg-cover sm:space-y-10 md:ml-60 transition-all"
            style="background-image: url(assets/farming.jpg);"
          >
            <div
              class="flex flex-col mx-10 space-y-5"
            >
              <h1>
                LandingPage.hero.title
              </h1>
              <p>
                LandingPage.hero.description
              </p>
              <div
                class="flex flex-col space-y-5 sm:flex-row sm:space-y-0 space-x-0 sm:space-x-5"
              >
                <a
                  class="primary"
                  href=""
                >
                  LandingPage.hero.primaryButtonLabel
                </a>
                <a
                  class="secondary"
                  href=""
                >
                  LandingPage.hero.secondaryButtonTitleLabel
                </a>
              </div>
            </div>
          </section>
          <!--container-->
        </app-hero>
        <!--container-->
        <app-frame
          id="products"
          ng-reflect-vm="[object Object]"
        >
          <section
            class="flex flex-col gap-y-10 py-20 mx-5 sm:mx-20 lg:mx-40"
          >
            <h2>
              LandingPage.frames.0.title
            </h2>
            <app-product-list>
              <div
                class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-8 mt-10 transition-all"
              >
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="products,1"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.0.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.0.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.0.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="products,2"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.1.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.1.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.1.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="products,3"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.2.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.2.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.2.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <!--container-->
              </div>
            </app-product-list>
          </section>
          <!--container-->
        </app-frame>
        <!--ng-container-->
        <app-frame
          id="partners"
          ng-reflect-vm="[object Object]"
        >
          <section
            class="flex flex-col gap-y-10 py-20 mx-5 sm:mx-20 lg:mx-40"
          >
            <h2>
              LandingPage.frames.1.title
            </h2>
            <app-partner-list>
              <div
                class="flex flex-wrap flex-row justify-center ease-in-out transition-all"
              >
                <!--container-->
                <app-partner-logo
                  ng-reflect-vm="[object Object]"
                >
                  <img
                    alt="LandingPage.partners.0.altText"
                    class="w-52 h-52"
                    src="assets/lepke.jpg"
                  />
                </app-partner-logo>
                <!--container-->
                <!--container-->
                <app-partner-logo
                  ng-reflect-vm="[object Object]"
                >
                  <img
                    alt="LandingPage.partners.0.altText"
                    class="w-52 h-52"
                    src="assets/lepke.jpg"
                  />
                </app-partner-logo>
                <!--container-->
                <!--container-->
                <app-partner-logo
                  ng-reflect-vm="[object Object]"
                >
                  <img
                    alt="LandingPage.partners.0.altText"
                    class="w-52 h-52"
                    src="assets/lepke.jpg"
                  />
                </app-partner-logo>
                <!--container-->
                <!--container-->
                <app-partner-logo
                  ng-reflect-vm="[object Object]"
                >
                  <img
                    alt="LandingPage.partners.0.altText"
                    class="w-52 h-52"
                    src="assets/lepke.jpg"
                  />
                </app-partner-logo>
                <!--container-->
                <!--container-->
                <app-partner-logo
                  ng-reflect-vm="[object Object]"
                >
                  <img
                    alt="LandingPage.partners.0.altText"
                    class="w-52 h-52"
                    src="assets/lepke.jpg"
                  />
                </app-partner-logo>
                <!--container-->
                <!--container-->
              </div>
            </app-partner-list>
            <!--container-->
            <swiper-container
              autoplay-delay="2500"
              autoplay-disable-on-interaction="false"
              class="w-full sm:w-2/3 md:w-1/2"
              keyboard="true"
              pagination="true"
              pagination-clickable="true"
              pagination-dynamic-bullets="true"
              space-between="30"
            >
              <swiper-slide
                class="pb-10"
              >
                <app-testimonial-item
                  ng-reflect-vm="[object Object]"
                >
                  <div
                    class="p-6 full mx-auto text-center rounded-md shadow transition-all cursor-pointer"
                  >
                    <p
                      class="text-lg italic text-gray-700"
                    >
                      "LandingPage.testimonials.0.message"
                    </p>
                    <p
                      class="font-semibold text-gray-900 mt-4"
                    >
                      John Doe
                    </p>
                    <p
                      class="text-gray-500"
                    >
                      LandingPage.testimonials.0.roleAndCompany
                    </p>
                  </div>
                  <!--container-->
                </app-testimonial-item>
              </swiper-slide>
              <swiper-slide
                class="pb-10"
              >
                <app-testimonial-item
                  ng-reflect-vm="[object Object]"
                >
                  <div
                    class="p-6 full mx-auto text-center rounded-md shadow transition-all cursor-pointer"
                  >
                    <p
                      class="text-lg italic text-gray-700"
                    >
                      "LandingPage.testimonials.1.message"
                    </p>
                    <p
                      class="font-semibold text-gray-900 mt-4"
                    >
                      Jane Smith
                    </p>
                    <p
                      class="text-gray-500"
                    >
                      LandingPage.testimonials.1.roleAndCompany
                    </p>
                  </div>
                  <!--container-->
                </app-testimonial-item>
              </swiper-slide>
              <swiper-slide
                class="pb-10"
              >
                <app-testimonial-item
                  ng-reflect-vm="[object Object]"
                >
                  <div
                    class="p-6 full mx-auto text-center rounded-md shadow transition-all cursor-pointer"
                  >
                    <p
                      class="text-lg italic text-gray-700"
                    >
                      "LandingPage.testimonials.2.message"
                    </p>
                    <p
                      class="font-semibold text-gray-900 mt-4"
                    >
                      Emily Johnson
                    </p>
                    <p
                      class="text-gray-500"
                    >
                      LandingPage.testimonials.2.roleAndCompany
                    </p>
                  </div>
                  <!--container-->
                </app-testimonial-item>
              </swiper-slide>
              <swiper-slide
                class="pb-10"
              >
                <app-testimonial-item
                  ng-reflect-vm="[object Object]"
                >
                  <div
                    class="p-6 full mx-auto text-center rounded-md shadow transition-all cursor-pointer"
                  >
                    <p
                      class="text-lg italic text-gray-700"
                    >
                      "LandingPage.testimonials.3.message"
                    </p>
                    <p
                      class="font-semibold text-gray-900 mt-4"
                    >
                      Emily Johnson
                    </p>
                    <p
                      class="text-gray-500"
                    >
                      LandingPage.testimonials.3.roleAndCompany
                    </p>
                  </div>
                  <!--container-->
                </app-testimonial-item>
              </swiper-slide>
              <swiper-slide
                class="pb-10"
              >
                <app-testimonial-item
                  ng-reflect-vm="[object Object]"
                >
                  <div
                    class="p-6 full mx-auto text-center rounded-md shadow transition-all cursor-pointer"
                  >
                    <p
                      class="text-lg italic text-gray-700"
                    >
                      "LandingPage.testimonials.4.message"
                    </p>
                    <p
                      class="font-semibold text-gray-900 mt-4"
                    >
                      Emily Johnson
                    </p>
                    <p
                      class="text-gray-500"
                    >
                      LandingPage.testimonials.4.roleAndCompany
                    </p>
                  </div>
                  <!--container-->
                </app-testimonial-item>
              </swiper-slide>
              <swiper-slide
                class="pb-10"
              >
                <app-testimonial-item
                  ng-reflect-vm="[object Object]"
                >
                  <div
                    class="p-6 full mx-auto text-center rounded-md shadow transition-all cursor-pointer"
                  >
                    <p
                      class="text-lg italic text-gray-700"
                    >
                      "LandingPage.testimonials.5.message"
                    </p>
                    <p
                      class="font-semibold text-gray-900 mt-4"
                    >
                      Emily Johnson
                    </p>
                    <p
                      class="text-gray-500"
                    >
                      LandingPage.testimonials.5.roleAndCompany
                    </p>
                  </div>
                  <!--container-->
                </app-testimonial-item>
              </swiper-slide>
              <!--container-->
            </swiper-container>
            <!--container-->
          </section>
          <!--container-->
        </app-frame>
        <!--ng-container-->
        <!--container-->
        <!--container-->
        <!--container-->
        <!--container-->
        <!--container-->
      </app-landing-page>
      <!--container-->
    </div>
  </div>
</div>
`;

exports[`AppComponent should navigate to "/products" and render the ProductsPageComponent 1`] = `
<div
  id="root5"
  ng-version="18.2.8"
>
  <div
    class="min-h-screen flex flex-col"
  >
    <app-header>
      <section
        class="p-4 flex justify-center"
      >
        <div
          class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
        >
          <app-logo
            ng-reflect-vm="[object Object]"
          >
            <a
              href=""
            >
              <img
                alt="logo"
                class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
                src="assets/phoenix.jpg"
              />
            </a>
          </app-logo>
          <app-nav>
            <nav
              class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
            >
              <!--container-->
            </nav>
          </app-nav>
        </div>
      </section>
    </app-header>
    <!--container-->
    <div
      class="flex-grow"
    >
      <router-outlet />
      <app-products-page>
        <app-frame
          ng-reflect-vm="[object Object]"
        >
          <section
            class="flex flex-col gap-y-10 py-20 mx-5 sm:mx-20 lg:mx-40"
          >
            <h2>
              LandingPage.frames.0.title
            </h2>
            <app-product-list>
              <div
                class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-8 mt-10 transition-all"
              >
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="1"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.0.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.0.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.0.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="2"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.1.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.1.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.1.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="3"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.2.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.2.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.2.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <!--container-->
              </div>
            </app-product-list>
          </section>
          <!--container-->
        </app-frame>
        <!--container-->
      </app-products-page>
      <!--container-->
    </div>
  </div>
</div>
`;

exports[`AppComponent should navigate to "/products/id" and render the ProductsPageComponent 1`] = `
<div
  id="root6"
  ng-version="18.2.8"
>
  <div
    class="min-h-screen flex flex-col"
  >
    <app-header>
      <section
        class="p-4 flex justify-center"
      >
        <div
          class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
        >
          <app-logo
            ng-reflect-vm="[object Object]"
          >
            <a
              href=""
            >
              <img
                alt="logo"
                class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
                src="assets/phoenix.jpg"
              />
            </a>
          </app-logo>
          <app-nav>
            <nav
              class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
            >
              <!--container-->
            </nav>
          </app-nav>
        </div>
      </section>
    </app-header>
    <!--container-->
    <div
      class="flex-grow"
    >
      <router-outlet />
      <app-products-page>
        <app-frame
          ng-reflect-vm="[object Object]"
        >
          <section
            class="flex flex-col gap-y-10 py-20 mx-5 sm:mx-20 lg:mx-40"
          >
            <h2>
              LandingPage.frames.0.title
            </h2>
            <app-product-list>
              <div
                class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-8 mt-10 transition-all"
              >
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="1"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.0.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.0.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.0.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="2"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.1.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.1.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.1.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <app-product-item
                  class="h-full"
                  ng-reflect-router-link="3"
                  ng-reflect-vm="[object Object]"
                  tabindex="0"
                >
                  <a
                    class="cursor-pointer block p-4 w-full h-full rounded-md shadow-md hover:shadow-xl transition-all"
                  >
                    <img
                      alt="LandingPage.products.2.title"
                      class="w-auto h-auto object-cover rounded-md"
                      src="assets/lepke.jpg"
                    />
                    <h3
                      class="font-bold text-xl sm:text-2xl mt-4"
                    >
                      LandingPage.products.2.title
                    </h3>
                    <p
                      class="mt-4"
                    >
                      LandingPage.products.2.description
                    </p>
                  </a>
                  <!--container-->
                </app-product-item>
                <!--container-->
              </div>
            </app-product-list>
          </section>
          <!--container-->
        </app-frame>
        <!--container-->
      </app-products-page>
      <!--container-->
    </div>
  </div>
</div>
`;

exports[`AppComponent should not render <app-nav-item> when there is no item provided 1`] = `
<div
  id="root1"
  ng-version="18.2.8"
>
  <div
    class="min-h-screen flex flex-col"
  >
    <app-header>
      <section
        class="p-4 flex justify-center"
      >
        <div
          class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
        >
          <app-logo
            ng-reflect-vm="[object Object]"
          >
            <a
              href=""
            >
              <img
                alt="logo"
                class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
                src="assets/phoenix.jpg"
              />
            </a>
          </app-logo>
          <app-nav>
            <nav
              class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
            >
              <!--container-->
            </nav>
          </app-nav>
        </div>
      </section>
    </app-header>
    <!--container-->
    <div
      class="flex-grow"
    >
      <router-outlet />
      <!--container-->
    </div>
  </div>
</div>
`;

exports[`AppComponent should not render the template when the VM is not provided 1`] = `
<div
  id="root0"
  ng-version="18.2.8"
>
  <div
    class="min-h-screen flex flex-col"
  >
    <!--container-->
    <div
      class="flex-grow"
    >
      <router-outlet />
      <!--container-->
    </div>
  </div>
</div>
`;

exports[`AppComponent should render 1 <app-nav-item> when 1 item is provided 1`] = `
<div
  id="root2"
  ng-version="18.2.8"
>
  <div
    class="min-h-screen flex flex-col"
  >
    <app-header>
      <section
        class="p-4 flex justify-center"
      >
        <div
          class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
        >
          <app-logo
            ng-reflect-vm="[object Object]"
          >
            <a
              href=""
            >
              <img
                alt="logo"
                class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
                src="assets/phoenix.jpg"
              />
            </a>
          </app-logo>
          <app-nav>
            <nav
              class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
            >
              <app-nav-item
                ng-reflect-vm="[object Object]"
              >
                <a
                  class="sm:hover:text-gray-400 duration-150 cursor-pointer"
                  href=""
                  rel="null"
                  target="null"
                >
                  AppComponent.navItems.0.title
                </a>
              </app-nav-item>
              <!--container-->
            </nav>
          </app-nav>
        </div>
      </section>
    </app-header>
    <!--container-->
    <div
      class="flex-grow"
    >
      <router-outlet />
      <!--container-->
    </div>
  </div>
</div>
`;

exports[`AppComponent should render 5 <app-nav-item> in order when 5 items are provided 1`] = `
<div
  id="root3"
  ng-version="18.2.8"
>
  <div
    class="min-h-screen flex flex-col"
  >
    <app-header>
      <section
        class="p-4 flex justify-center"
      >
        <div
          class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
        >
          <app-logo
            ng-reflect-vm="[object Object]"
          >
            <a
              href=""
            >
              <img
                alt="logo"
                class="w-12 h-12 md:w-16 md:h-16 mb-4 md:mb-0"
                src="assets/phoenix.jpg"
              />
            </a>
          </app-logo>
          <app-nav>
            <nav
              class="flex flex-col items-center space-y-4 text-center md:items-start md:flex-row md:space-y-0 md:space-x-10 md:text-left"
            >
              <app-nav-item
                ng-reflect-vm="[object Object]"
              >
                <a
                  class="sm:hover:text-gray-400 duration-150 cursor-pointer"
                  href=""
                  rel="null"
                  target="null"
                >
                  AppComponent.navItems.0.title
                </a>
              </app-nav-item>
              <app-nav-item
                ng-reflect-vm="[object Object]"
              >
                <a
                  class="sm:hover:text-gray-400 duration-150 cursor-pointer"
                  href=""
                  rel="null"
                  target="null"
                >
                  AppComponent.navItems.1.title
                </a>
              </app-nav-item>
              <app-nav-item
                ng-reflect-vm="[object Object]"
              >
                <a
                  class="sm:hover:text-gray-400 duration-150 cursor-pointer"
                  href=""
                  rel="null"
                  target="null"
                >
                  AppComponent.navItems.2.title
                </a>
              </app-nav-item>
              <app-nav-item
                ng-reflect-vm="[object Object]"
              >
                <a
                  class="sm:hover:text-gray-400 duration-150 cursor-pointer"
                  href=""
                  rel="null"
                  target="null"
                >
                  AppComponent.navItems.3.title
                </a>
              </app-nav-item>
              <app-nav-item
                ng-reflect-vm="[object Object]"
              >
                <a
                  class="sm:hover:text-gray-400 duration-150 cursor-pointer"
                  href=""
                  rel="null"
                  target="null"
                >
                  AppComponent.navItems.1.title
                </a>
              </app-nav-item>
              <!--container-->
            </nav>
          </app-nav>
        </div>
      </section>
    </app-header>
    <!--container-->
    <div
      class="flex-grow"
    >
      <router-outlet />
      <!--container-->
    </div>
  </div>
</div>
`;
